internal/scaffold/cmd.go

```diff
@@ -203,7 +203,16 @@ func addMetrics(ctx context.Context, cfg *rest.Config, namespace string) {
 	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
 	// necessary to configure Prometheus to scrape metrics from this operator.
 	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+
+	// Get the namespace the operator is currently deployed in.
+	operatorNs, err := k8sutil.GetOperatorNamespace()
```

Comment:
@camilamacedo86 - Shouldn't https://github.com/operator-framework/operator-sdk/pull/2601/files#diff-6438d7f67aed0d3fd6727a1f27133faeR237 be updated to invoke `k8sutil.GetWatchNamespace()` if the CRD is cluster-scoped and `WATCH_NAMESPACE` is set to `""`? When using cluster-scoped CRDs, the CRs themselves don't live in the namespace returned by `k8sutil.OperatorNamespace()`, right?

Reply:
By default, the metrics service is created in the same namespace where the operator is deployed. See: `service/memcached-operator-metrics `

```
$ kubectl get all -n memcached 
NAME                                      READY   STATUS    RESTARTS   AGE
pod/example-memcached-7c4df9b7b4-lzd6j    1/1     Running   0          64s
pod/example-memcached-7c4df9b7b4-wbtkz    1/1     Running   0          64s
pod/example-memcached-7c4df9b7b4-wt6jb    1/1     Running   0          64s
pod/memcached-operator-56f54d84bf-zrtfv   1/1     Running   0          69s

NAME                                 TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
service/example-memcached            ClusterIP   10.108.124.47   <none>        11211/TCP           63s
service/memcached-operator-metrics   ClusterIP   10.108.67.82    <none>        8383/TCP,8686/TCP   66s

NAME                                 READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/example-memcached    3/3     3            3           65s
deployment.apps/memcached-operator   1/1     1            1           70s

NAME                                            DESIRED   CURRENT   READY   AGE
replicaset.apps/example-memcached-7c4df9b7b4    3         3         3       65s
replicaset.apps/memcached-operator-56f54d84bf   1         1         1       70s
```

I think you are trying to raise here the scenario where the `service/CR` should be created in the namespace where the CR is applied. Am I right?  Could you please open an issue with your scenario and ping me?

Reply:
See; https://github.com/operator-framework/operator-sdk/issues/2707

Reply:
No. I'm not talking about the creation of the actual Kubernetes Service and ServiceMonitor resources. I'm talking about the way in which the Go code that serves the custom resource metrics tracks which custom resources to report on. After filtering the GVKs, `serveCRMetrics` restricts which namespaces should be searched for custom resources:

```go
	// Get the namespace the operator is currently deployed in.
	operatorNs, err := k8sutil.GetOperatorNamespace()
	if err != nil {
		return err
	}
	// To generate metrics in other namespaces, add the values below.
	ns := []string{operatorNs}
	// Generate and serve custom resource specific metrics.
	err = kubemetrics.GenerateAndServeCRMetrics(cfg, ns, filteredGVK, metricsHost, operatorMetricsPort)
	if err != nil {
		return err
	}
```

The call to `k8sutil.GetOperatorNamespace()`, returns the name of the namespace in which the operator is running, but cluster-scoped resources don't belong to a namespace. I think the call to `k8sutil.GetOperatorNamespace()` need to be replaced with a call to `k8sutil. GetWatchNamespace()`. Only then will the namespace against which to filter CRs be correctly set to `""` when using cluster-scoped CRDs

Reply:
HI @robbie-demuth, 

Note that it was not really changed in this PR. Before that it was using the OperatorNamespace. See 
https://github.com/camilamacedo86/operator-sdk/blob/ad1d4d1953abf22c336f67198fc23a489c0f4fdb/internal/scaffold/cmd.go#L213 

So, let's discuss your suggestion and check it via issue #2707
c/c @estroz

Reply:
Understood. I just commented here because this PR was listed as a fix for https://github.com/operator-framework/operator-sdk/issues/1858 and, without the update to `serveCRMetrics`, the issue isn't really resolved

Reply:
Note that to solve issue #1858 can cause by many raisons and we addressed the diff scenarios in more than one pr. If you check the current scaffold you will see: 

```
         // The metrics will be generated from the namespaces which are returned here.
	// NOTE that passing nil or an empty list of namespaces in GenerateAndServeCRMetrics will result in an error.
	ns, err := kubemetrics.GetNamespacesForMetrics(operatorNs)
	if err != nil {
		return err
	}
	// Generate and serve custom resource specific metrics.
	err = kubemetrics.GenerateAndServeCRMetrics(cfg, ns, filteredGVK, metricsHost, operatorMetricsPort)
	if err != nil {
		return err
	}

```

Then, it has a clear msg over what is expected in this place. `NOTE that passing nil or an empty list of namespaces in GenerateAndServeCRMetrics will result in an error.` and then, because of this, we cannot use the WATCH_NAMESPACE in this point.

Reply:
I've updated one of my projects to call `k8sutil.GetWatchNamespace()` instead of `k8sutil.GetOperatorNamespace()` and it does work. It's important to note that if `WATCH_NAMESPACE` is set to `""`, the list of namespaces passed to `GenerateAndServeCRMetrics` won't be nil or empty. Instead it'll be equivalent to `[]string{""}`